scripts/upgrade-assure/README.md

```diff
@@ -1,75 +1,132 @@
 
-You can find the latest AviaOne testnet snapshot [here](https://aviaone.com/blockchains-service/elystestnet-1-elys.html#8).
+Download the specific binary for Darwin ARM64 architecture using `curl`:
 
-```
-make install
-go run ./scripts/upgrade-assure/... https://services.elystestnet-1.elys.aviaone.com/elystestnet-1_2024-03-06.tar.gz ~/go/bin/elysd ~/go/bin/elysd --skip-proposal
+```bash
+curl -L https://github.com/elys-network/elys/releases/download/v0.29.31/elysd-v0.29.31-darwin-arm64 -o /tmp/elysd-v0.29.31
 ```
 
-# How can I perform a test with a version upgrade that involves extensive changes to data structures?
+### Step 4: Retrieve Testnet Snapshot
 
-```
-git checkout v0.29.30
-make install
-cp -a ~/go/bin/elysd /tmp/elysd-v0.29.30
-```
+Fetch the latest testnet snapshot, necessary for data migration using `curl`:
 
-```
-go run ./scripts/upgrade-assure/... --home /tmp/elys --home2 /tmp/elysd2 https://tools.highstakes.ch/files/elys.tar.gz /tmp/elysd-v0.29.30 /tmp/elysd-v0.29.31 --skip-node-start
+```bash
+curl -L https://snapshots-testnet.stake-town.com/elys/elystestnet-1_latest.tar.lz4 -o /tmp/snapshot.tar.lz4
 ```
 
-```
+### Step 5: Checkout the Previous Version
+
+Switch to the previously stable version:
+
+```bash
 git checkout v0.29.31
-make install
-cp -a ~/go/bin/elysd /tmp/elysd-v0.29.31
 ```
 
+### Step 6: Run Upgrade-Assure Script
+
+#### 6a: Initial Run
+
+Run the upgrade-assure script without starting the node:
+
+```bash
+go run ./scripts/upgrade-assure/... /tmp/snapshot.tar.lz4 /tmp/elysd-v0.29.31 ~/go/bin/elysd --skip-node-start
 ```
-go run ./scripts/upgrade-assure/... --home /tmp/elys --home2 /tmp/elysd2 https://tools.highstakes.ch/files/elys.tar.gz /tmp/elysd-v0.29.30 /tmp/elysd-v0.29.31 --skip-snapshot --skip-chain-init
-```
-# Troubleshooting
 
-These are some problems and its solutions when the script is executed
+#### 6b: Handle Potential Errors
+
+Address any type errors, such as difficulties in unmarshaling strings into integer fields in Go struct fields.
+
+#### 6c: Update the Script
+
+Modify `scripts/upgrade-assure/types.go` to reflect data structure changes necessary to resolve type errors.
+
+#### 6d: Retry Upgrade-Assure
+
+Repeat the process after updating the script:
 
-- The O.S could kill the process if you don´t have enough ram this happened in 6.6.24-1-MANJARO (64 bits) with 16gb of ram a solution was create a swapFile
+```bash
+go run ./scripts/upgrade-assure/... /tmp/snapshot.tar.lz4 /tmp/elysd-v0.29.31 ~/go/bin/elysd --skip-node-start
 ```
-https://wiki.manjaro.org/index.php?title=Swap#Using_a_Swapfile
+
+### Step 7: Checkout to Latest Changes Branch
+
+Switch back to the main branch to incorporate the latest changes:
+
+```bash
+git checkout main
 ```
--  Timeout, if you have timeout problems with the node or waiting for the next block you can use these flags: 
-timeout-wait-for-node is expressed in seconds
-timeout-next-block is expressed in minutes
 
+### Step 8: Final Upgrade Command
```

Comment:
Should we include here the steps to run both alice and bob nodes?

Reply:
@Wgil both alice and bob nodes are running automatically when running this command